enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+
+This means that for a Machine, we expect the authoritative controller to perform the actual deletion work within the infrastructure provider.
+Once it has completed this effort, it will remove its own Finalizer.
+
+As soon as the synchronisation controller notices the Machine has been deleted, it should ensure that the Machine API/Cluster API equivalent (if present) is also deleted.
+It should then wait until the authoritative Machine controller removes its finalizer before removing both its own, and the Machine API/Cluster API Machine finalizer if present.
+In scenarios where both Machine API and Cluster API have been authoritative, it is expected that both Machine controllers will have added their own Finalizers, therefore we expect the synchronisation controller should be able to handle this.
+
+It is also feasible that a customer may want to remove the Machine API resources after they have migrated to Cluster API and no longer require the Machine API synchronisation.
+To allow for this, if a non-authoritative Machine API parent[^8] resource is deleted, the deletion event will not be synchronised to the Cluster API equivalent and the synchronisation controller will ensure finalizers are removed as appropriate.
+In this scenario, the Cluster API equivalent will continue to operate on its own.
+To ensure users are aware of this situation, webhooks will be introduced to warn users that the deletion will not be synchronised.
+
+[^8]: To allow continued operation of MachineSets, ControlPlaneMachineSets and MachineHealthChecks, if a Machine API Machine that is owned is deleted,
+then the deletion event is synchronised to Cluster API.
+This will allow for a MachineSet/ControlPlaneMachineSet to own mixed instances (i.e. those managed by both Machine API and Cluster API), and continue to support higher level remediation automation.
+To remove the Machine API resources, the user should delete the parent, i.e. the MachineSet or ControlPlaneMachineSet on the Machine API side.
+
+#### Scenarios with mixed Machines
+
+The following scenarios explain the logical flow of what must happen if any of the controllers are monitoring a selection of Machines where the authoritative API is mixed between Cluster API and Machine API.
+
+When the selection of Machines all match the same API, the normal flows will be observed.
+
+##### MachineSet scale up
+
+If a MachineSet scales up, it will create a new Machine as it does today.
+Because the managed resource in this scenario is the MachineSet the synchronisation controller will ensure that the Machine resources are reflected in both Machine API and Cluster API.
+
+If the MachineSet is a Cluster API MachineSet and there is no Machine API equivalent, this synchronisation will not happen and the machines will be created as Cluster API only.
+
+New Machines created in the scale up with default to the same authoritative API as the MachineSet that creates them.
```

Comment:
Yes, I don't think it will be an issue, the sync controller should handle this, but it is a behavioural change we will likely want to document as part of this, CC @jeana-redhat